mono/metadata/sre.c

```diff
@@ -1547,7 +1549,21 @@ mono_reflection_type_get_handle (MonoReflectionType* ref, MonoError *error)
 				return NULL;
 			}
 		}
-
+		/* Need to resolve the generic_type in order for it to create its generic context. */
```

Comment:
The purpose of the recent SRE changes was to delay the creation of the MonoClass corresponding to the TypeBuilder until CreateType () is called, so the runtime wouldn't have to deal with half created types. This change moves in the opposite direction. Does the testcase work on windows ?